Mathlib/LinearAlgebra/Matrix/Spectrum.lean

```diff
@@ -53,139 +50,135 @@ noncomputable def eigenvectorBasis : OrthonormalBasis n 𝕜 (EuclideanSpace 
+/--Unitary matrix whose columns are Orthonormal Basis of Eigenvectors of Hermitian Matrix-/
+noncomputable def eigenvectorUnitary {𝕜 : Type*} [RCLike 𝕜] {n : Type*}
+    [Fintype n]{A : Matrix n n 𝕜} [DecidableEq n] (hA : Matrix.IsHermitian A) :
+    Matrix.unitaryGroup n 𝕜 :=
+  ⟨(EuclideanSpace.basisFun n 𝕜).toBasis.toMatrix (hA.eigenvectorBasis).toBasis,
+    (EuclideanSpace.basisFun n 𝕜).toMatrix_orthonormalBasis_mem_unitary (eigenvectorBasis hA)⟩
+#align matrix.is_hermitian.eigenvector_matrix Matrix.IsHermitian.eigenvectorUnitary
+
+lemma eigenvectorUnitary_coe {𝕜 : Type*} [RCLike 𝕜] {n : Type*} [Fintype n]
+    {A : Matrix n n 𝕜} [DecidableEq n] (hA : Matrix.IsHermitian A) :
+    eigenvectorUnitary hA =
+      (EuclideanSpace.basisFun n 𝕜).toBasis.toMatrix (hA.eigenvectorBasis).toBasis :=
+  rfl
+
+@[simp]
+theorem eigenvectorUnitary_apply (i j : n) :
+    eigenvectorUnitary hA i j = ⇑(hA.eigenvectorBasis j) i :=
+  rfl
+#align matrix.is_hermitian.eigenvector_matrix_apply Matrix.IsHermitian.eigenvectorUnitary_apply
+
+theorem eigenvectorUnitary_mulVec (j : n) :
+    eigenvectorUnitary hA *ᵥ Pi.single j 1 = ⇑(hA.eigenvectorBasis j) := by
+  simp only [mulVec_single, eigenvectorUnitary_apply, mul_one]
+
+theorem star_eigenvectorUnitary_mulVec (j : n) :
+    (star (eigenvectorUnitary hA : Matrix n n 𝕜)) *ᵥ ⇑(hA.eigenvectorBasis j) = Pi.single j 1 := by
+  rw [← eigenvectorUnitary_mulVec, mulVec_mulVec, unitary.coe_star_mul_self, one_mulVec]
 
 /-- **Diagonalization theorem**, **spectral theorem** for matrices; A hermitian matrix can be
 diagonalized by a change of basis.
 
 For the spectral theorem on linear maps, see
 `LinearMap.IsSymmetric.eigenvectorBasis_apply_self_apply`. -/
+theorem star_mul_self_mul_eq_diagonal :
+    (star (eigenvectorUnitary hA : Matrix n n 𝕜)) * A * (eigenvectorUnitary hA : Matrix n n 𝕜)
+      = diagonal (RCLike.ofReal ∘ hA.eigenvalues) := by
+  apply Matrix.toEuclideanLin.injective
+  apply Basis.ext (EuclideanSpace.basisFun n 𝕜).toBasis
+  intro i
+  simp only [toEuclideanLin_apply, OrthonormalBasis.coe_toBasis, EuclideanSpace.basisFun_apply,
+    WithLp.equiv_single, ← mulVec_mulVec, eigenvectorUnitary_mulVec, ← mulVec_mulVec,
+    mulVec_eigenvectorBasis, Matrix.diagonal_mulVec_single, mulVec_smul,
+    star_eigenvectorUnitary_mulVec, RCLike.real_smul_eq_coe_smul (K := 𝕜), WithLp.equiv_symm_smul,
+    WithLp.equiv_symm_single, Function.comp_apply, mul_one, WithLp.equiv_symm_single]
+  apply PiLp.ext
+  intro j
+  simp only [PiLp.smul_apply, EuclideanSpace.single_apply, smul_eq_mul, mul_ite, mul_one, mul_zero]
+#align matrix.is_hermitian.star_mul_self_mul_eq_diagonal Matrix.IsHermitian.star_mul_self_mul_eq_diagonal
+
+/-- *spectral theorem* A hermitian matrix can be can be
+replaced by a diagonal matrix sandwiched between the eigenvector unitaries. This alternate form
+allows direct rewriting of A since: <| A = V D V⁻¹$ -/
```

Comment:
Can you update this docstring to be more like the one above? Especially since this is the thing we're calling `spectral_theorem`